Automattic/wp-calypso
# Staging Sites: Ensure users can't transfer domain to a staging site

Related to https://github.com/Automattic/dotcom-forge/issues/2077

## Proposed Changes

This PR eliminates the ability for users to transfer their domain to a staging site, providing greater control and security over domain transfers, and ensuring that such transfers can no longer occur.

## Testing Instructions



### Test transfer domain

#### Test 1
1. Create a new site ( with a domain. It's suggested to buy a new `.blog` domain)
2. Navigate to `/domains/manage/<your-site>/transfer/other-site/<your domain>`
3. Search for a staging site and ensure that none of them appears in the list.

| **Before** | **After** |
| :---:         |    :---:       |
| ![transfer1-before](https://user-images.githubusercontent.com/497103/228844054-b7a23e8d-53c3-4f9a-8e63-fbc696464900.png)| ![transfer1-after](https://user-images.githubusercontent.com/497103/228844089-6f768112-2d6d-4587-a15f-f8f635a59a76.png) |

#### Test 2
1. Create a new site ( with a domain. It's suggested to buy a new `.blog` domain)
2. Navigate to `/domains/add`
3. Search for a staging site and ensure that none of them appears in the list.

| **Before** | **After** |
| :---:         |    :---:       |
| ![transfer2-before](https://user-images.githubusercontent.com/497103/228847298-7a8d570f-4bc8-4668-b93b-52acd9e97a6b.png)| ![transfer2-after](https://user-images.githubusercontent.com/497103/228847322-d6bdd760-3bab-4ad1-879a-2c55b0a87219.png)|

**NOTE:** The change mentioned on **Test 2**, also applies to:
- `/domains/add/mapping`
- `/domains/add/transfer`
- `/domains/add/site-redirect`
## Pre-merge Checklist
